<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>698</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>stm32Downloader</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="SerialGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>161</width>
      <height>291</height>
     </rect>
    </property>
    <property name="title">
     <string>串口设置</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>串口号</string>
     </property>
    </widget>
    <widget class="QComboBox" name="PortBox">
     <property name="geometry">
      <rect>
       <x>68</x>
       <y>30</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
    </widget>
    <widget class="QComboBox" name="BaudRateBox">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>60</y>
       <width>69</width>
       <height>21</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>6</number>
     </property>
     <item>
      <property name="text">
       <string>2400</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4800</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>9600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>19200</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>38400</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>57600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>115200</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>波特率</string>
     </property>
    </widget>
    <widget class="QComboBox" name="DataBox">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>90</y>
       <width>69</width>
       <height>21</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>3</number>
     </property>
     <item>
      <property name="text">
       <string>5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>数据位</string>
     </property>
    </widget>
    <widget class="QComboBox" name="CheckBox">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>120</y>
       <width>69</width>
       <height>21</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>无校验</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>奇校验</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>偶校验</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>120</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>校验位</string>
     </property>
    </widget>
    <widget class="QComboBox" name="StopBox">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>150</y>
       <width>69</width>
       <height>21</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1.5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>停止位</string>
     </property>
    </widget>
    <widget class="QComboBox" name="FluidBox">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>180</y>
       <width>69</width>
       <height>21</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>无流控</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>硬件流</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>软件流</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>180</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>流控制</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="AsciiSelect">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>ASCII</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="HexSelected">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>210</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>HEX</string>
     </property>
    </widget>
    <widget class="QPushButton" name="ClearButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>260</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>清空显示区</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="TimeCheckBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>时间</string>
     </property>
    </widget>
    <widget class="QPushButton" name="SerialConnectButton">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>230</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>连接</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="DisplayGriup">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>0</y>
      <width>591</width>
      <height>381</height>
     </rect>
    </property>
    <property name="title">
     <string>串口显示区</string>
    </property>
    <widget class="QTextEdit" name="SerialScreenText">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>571</width>
       <height>321</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="SerialSendLine">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>350</y>
       <width>481</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="SerialSendButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>350</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>发  送</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="UserGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>161</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>当前用户信息</string>
    </property>
    <widget class="QLabel" name="UserNameLabel">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="UserNameLabel_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>用户名：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>54</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>用户类型：</string>
     </property>
    </widget>
    <widget class="QLabel" name="IsAdminLabel">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>50</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="binGroup">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>451</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>bin文件下载</string>
    </property>
    <widget class="QTableView" name="SqlShowView">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>80</y>
       <width>291</width>
       <height>171</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="UploadPathButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>上传路径</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="UpLoadPath">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>20</y>
       <width>361</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="DownloadPath">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>50</y>
       <width>361</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="DownloadPathButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>下载路径</string>
     </property>
    </widget>
    <widget class="QPushButton" name="UploadButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>上传文件</string>
     </property>
    </widget>
    <widget class="QPushButton" name="DownloadButton">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>80</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>下载文件</string>
     </property>
    </widget>
    <widget class="QPushButton" name="DeleteButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>131</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>删除所选数据库文件</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Download2stmButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>131</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>下载所选文件到单片机</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="SqlText">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QProgressBar" name="Download2stmBar">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="SocketGroup">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>390</y>
      <width>301</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>bin文件网络共享</string>
    </property>
    <widget class="QRadioButton" name="ServerRadioButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Server</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="ClientRadioButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Client</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>本机IP：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ServerIP">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>对方IP：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ClientIP">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>50</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>端口号：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="PortSpinBox">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>80</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="value">
      <number>8080</number>
     </property>
    </widget>
    <widget class="QPushButton" name="SocketConnectButton">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>80</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>点击连接</string>
     </property>
    </widget>
    <widget class="QPushButton" name="SocketSendButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>80</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>发送文件</string>
     </property>
    </widget>
    <widget class="QPushButton" name="SocketReceivePathButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>108</y>
       <width>81</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>接受文件地址</string>
     </property>
    </widget>
    <widget class="QProgressBar" name="SocketBar">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>110</y>
       <width>171</width>
       <height>21</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QLineEdit" name="SocketReceivePath">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>271</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QTextEdit" name="SocketTextOutput">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>271</width>
       <height>81</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
